File: src/RentCars.Domain/Vehicles/VehicleBlank.cs
using RentCars.Domain.Vehicles.Enums;

namespace RentCars.Domain.Vehicles;

public class VehicleBlank
{
    public Guid? Id { get; set; }
    public String? Brand { get; set; }
    public String? Model { get; set; }
    public Int32? YearOfManufacture { get; set; }
    public VehicleClass? VehicleClass { get; set; }
    public String? BodyColor { get; set; }
    public BodyType? BodyType { get; set; }
    public Int32? EnginePower { get; set; }
    public Double? EngineCapacity { get; set; }
    public FuelType? FuelType { get; set; }
    public WheelDrive? WheelDrive { get; set; }
    public TransmissionType? TransmissionType { get; set; }
    public Double? DayCost { get; set; }
    public Double? TwoFourDaysCost { get; set; }
    public Double? FourSevenDaysCost { get; set; }
    public Double? SevenFourteenDaysCost { get; set; }
    public Double? FourteenAndMoreDaysCost { get; set; }
    public VehiclePhotoBlank[] ExistPhotos { get; set; }
}

File: API/api.client/src/domain/vehicles/vehicleBlank.ts
import { BodyType } from "./enums/bodyType";
import { FuelType } from "./enums/fuelType";
import { TransmissionType } from "./enums/transmissionType";
import { VehicleClass } from "./enums/vehicleClass";
import { WheelDrive } from "./enums/wheelDrive";
import { Vehicle } from "./vehicle";
import { VehiclePhotoBlank } from "./vehiclePhotoBlank";

export class VehicleBlank {
    constructor(
        public id: string | null,
        public brand: string | null,
        public model: string | null,
        public yearOfManufacture: number | null,
        public vehicleClass: VehicleClass,
        public bodyColor: string | null,
        public bodyType: BodyType,
        public enginePower: number | null,
        public engineCapacity: number,
        public fuelType: FuelType,
        public wheelDrive: WheelDrive,
        public transmissionType: TransmissionType,
        public dayCost: number | null,
        public twoFourDaysCost: number | null,
        public fourSevenDaysCost: number | null,
        public sevenFourteenDaysCost: number | null,
        public fourteenAndMoreDaysCost: number | null,
        public existPhotos: VehiclePhotoBlank[]
    ) { }

    public static empty() {
        return new VehicleBlank(null, null, null, null, VehicleClass.Comfort, null,
            BodyType.Sedan, null, 2.5, FuelType.Gasoline,
            WheelDrive.AllWheelDrive, TransmissionType.Automatic,
            null, null, null, null, null, []);
    }

    public static fromDomain(vehicle: Vehicle): VehicleBlank {
        return new VehicleBlank(vehicle.id, vehicle.brand, vehicle.model, vehicle.yearOfManufacture,
            vehicle.vehicleClass, vehicle.bodyColor, vehicle.bodyType, vehicle.enginePower,
            vehicle.engineCapacity, vehicle.fuelType, vehicle.wheelDrive, vehicle.transmissionType, 
            vehicle.dayCost, vehicle.twoFourDaysCost, vehicle.fourSevenDaysCost, vehicle.sevenFourteenDaysCost, 
            vehicle.fourteenAndMoreDaysCost, VehiclePhotoBlank.fromDomain(vehicle.photos)
        )
    }
}